#!/usr/bin/env python
PACKAGE = "dobot_coordinate_transformer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("h_max", int_t, 0, "An Integer parameter", 255, 0, 255)
gen.add("h_min", int_t, 0, "An Integer parameter", 0, 0, 255)
gen.add("s_max", int_t, 0, "An Integer parameter", 233, 0, 255)
gen.add("s_min", int_t, 0, "An Integer parameter", 71, 0, 255)
gen.add("v_max", int_t, 0, "An Integer parameter", 255, 0, 255)
gen.add("v_min", int_t, 0, "An Integer parameter", 80, 0, 255)

size_enum = gen.enum([ gen.const("Small", int_t, 0, "A small constant"),
                  gen.const("Medium", int_t, 1, "A medium constant"),
                  gen.const("Large", int_t, 2, "A large constant"),
                  gen.const("ExtraLarge", int_t, 3, "An extra large constant") ],
                  "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "dobot_coordinate_transformer", "config"))
